### ISSUE TITLE
Method to get computed preference value?

### ISSUE THREAD

**Issue by lukewarlow** [WICG/web-preferences-api#7]
Do we need a new method to get the computed preference value?

Currently, you'd have to use `matchMedia` which works but isn't necessarily very ergonomic.

For example to listen to a change in the value you'd have to do multiple matchMedias for each potential value.

For most preferences you could infer based on only 1, but color-scheme is open ended and could include other values in future, and `(prefers-contrast)` has 3 potential values (technically 4 but this API doesn't really make sense for "custom").

**Comment by lukewarlow** [comment]
While writing https://demo.lukewarlow.dev/web-preferences-api/ it was quite apparent that the current mechanism using matchMedia really isn't very ergonomic.

While it's not necessary for a first pass at this API I do think it's worth considering.

**Comment by lukewarlow** [comment]
I could imagine the new PreferenceObject interface extending eventtarget and having an onchange so you can track when the override value changes, but also having a new attribute/function that allows you to get the "computed" value of the preference.

**Comment by lukewarlow** [comment]
In TypeScript terms something like

```ts
interface PreferenceObject extends EventTarget {
	// null means the preference is not overridden
 	readonly override: string | null;
	// The computed value of the preference, the override if set or the default if not. Name TBD
	readonly value: string;

	requestOverride(value: string): Promise<void>;
	clearOverride(): void;

	onchange: ((this: PreferenceObject, ev: Event) => any) | null;
}
```

**Comment by bramus** [comment]
I built a POC browser extension that allows changing light/dark mode on a per-site basis by the click of a button. The extension taps into the Web Preferences API for this: read and override the value of `navigator.preferences.colorScheme`.

https://github.com/WICG/web-preferences-api/assets/213073/b542edc4-aef9-4417-bd04-c90cd369643b

One of the features of the extension is that the icon in toolbar reflects the active state of the page: if the page is in dark mode, the icon will show a dark icon. If it’s in light mode, the icon will show a light icon.

Right now my implementation isn’t 100% accurate, as I rely on watching the `prefers-color-scheme` media query via `window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', e => … })` which doesn’t always trigger.

My OS is in Light Mode.
- If I override `navigator.preferences.colorScheme` to dark, the MQ listener triggers, so I can update the icon to dark.
- If I then override `navigator.preferences.colorScheme` to light, MQ listener triggers, so I can update the icon to dark.
- If I then clear the `navigator.preferences.colorScheme`, the MQ listener **DOES NOT** trigger because the page basically went from light (override) to light (system).

Mutatis mutandis for an OS in Dark Mode where the MQ listener won’t trigger from Dark (system) to Dark (override).

For this reason, a change listener on the PreferenceObject would be feasible, as I also expressed in https://x.com/bramus/status/1733263658543669574 before being pointed to this issue right here.
